src/hooks/useTypeColorToColorMatch.tsx

```diff
@@ -0,0 +1,65 @@
+import * as React from 'react';
+import useTheme from 'hooks/useTheme';
+import keys from 'lodash/keys';
+import pick from 'lodash/pick';
+import { flatPalette, generatedColorShades, mainTypes } from '../theme/palette';
+import { ColorShapeFromComponent } from '../utils/themeFunctions';
+
+const defaultContextData = {
+  typesShadesColor: {},
+};
+
+const TypeColorToColorMatchContext = React.createContext<{ typesShadesColor: TypesShadeAndColors }>(
+  // @ts-ignore
+  defaultContextData
+);
+const useTypeColorToColorMatch = () => {
+  return React.useContext<{ typesShadesColor: TypesShadeAndColors }>(TypeColorToColorMatchContext);
+};
+
+export type TypesShadeAndColors = Record<typeof mainTypes[number], ColorShapeFromComponent>;
+
+/**
+ * Apply for each type passed what color it is and shade (shade now is auto applied at 400)
+ * return e.g { primary: { shade: 400, color: 'orange'}, info: { shade: 400, color: 'darkBlue'}}
+ */
+const calculateTypesShadeAndColors = (
+  types: Record<typeof mainTypes[number], generatedColorShades>,
+  palette: flatPalette
+) =>
+  keys(types).reduce((typeAcc, key) => {
```

Comment:
Is there any other available solution for this one? It seems a little bit complex (multiple reduces). Also, It does not seem to be performant (log n^3 complexity) and readable. 

If we have no time to refactor it in this PR - merge it - we can open an issue and check alternatives in the future.

Reply:
This only runs one time, at the system initializing. I know its a bit weird is what we discuss but i am really open to fix it and i think now is the time. Let's discuss it a bit over chat.